test/patch.jl

```diff
@@ -13,6 +13,26 @@ function strip_lineno!(expr::Expr)
    return expr
 end
 
+# Test for nested modules
+module ModA
+
+module ModB
+
+abstract type AbstractFoo end
+
+struct Foo <: AbstractFoo
+    x::String
+end
+
+end
+
+bar(f::ModB.AbstractFoo) = println("blah")
+
+end
+
+import ModA
```

Comment:
For 0.7, this will need to be
```julia
if VERSION >= v"0.7.0-DEV.1877"
    import .ModA
    import .ModA: bar, ModB
else
    import ModA
    import ModA: bar, ModB
end
```
because of https://github.com/JuliaLang/julia/pull/23579